File: reference/rfid.sdk.win.V02.09/rfid_reader/HF/samples/delphi/BufferModeSample/rfidlib_reader.pas
unit rfidlib_reader;
{
   Only for win32 platform
}
interface
uses Windows;


type
RFIDLIB_EVENTC_CB = procedure(wParam: longword; lParam: longword)  ; stdcall;
PBYTE=^byte  ;


function   RDR_GetLibVersion( buf:pchar ; nSize:longword):longword  ; stdcall ; external 'rfidlib_reader.dll' ;

function RDR_LoadReaderDrivers( path:string):integer ; stdcall ;external 'rfidlib_reader.dll'
function RDR_GetLoadedReaderDriverCount() :longword ; stdcall ;external 'rfidlib_reader.dll'
function RDR_GetLoadedReaderDriverOpt( idx:longword  ; option:string ; valueBuffer:pchar ;var nSize:longword ):integer ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_GetLoadedReaderDriverOptByName( ID:string ; option:string;  valueBuffer:pchar ;var nSize:longword) :integer ; stdcall ;external 'rfidlib_reader.dll'
function  COMPort_Enum():longword ; stdcall ;external 'rfidlib_reader.dll'
function  COMPort_GetEnumItem( idx:longword ; nameBuf:pchar ; nSize:longword ) :integer ; stdcall ;external 'rfidlib_reader.dll'

function  HID_Enum( DevId:string):longword ; stdcall ;external 'rfidlib_reader.dll'
function  HID_GetEnumItem( indx:longword ; infType:byte ; infBuffer:pchar ;var nSize:longword ):integer ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_Open( connStr:string ;var hrOut:longword):integer ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_Close(hr:longword) :integer ; stdcall ;external 'rfidlib_reader.dll'

function   RDR_CreateInvenParamSpecList():longword  ; stdcall ; external 'rfidlib_reader.dll' ;

function   RDR_TagInventory( hr:longword ;AIType:byte;AntennaCoun:byte;AntennaIDs:PBYTE;InvenParamSpecList:longword):integer; stdcall ; external 'rfidlib_reader.dll' ;

function   RDR_GetTagDataReportCount( hr:longword) :longword; stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_GetTagDataReport( hr:longword; seek:byte):longword ;stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_EnableAsyncTagReportOutput(hr:longWord ;AsycType:byte; msg:longword ; hwnd:HWND ;cb:RFIDLIB_EVENTC_CB):integer ;stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_DisableAsyncTagReportOutput( hr:longword):integer ;stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_TagDisconnect( hr:longword; hTag:longword):integer;  stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_GetReaderLastReturnError( hr:longword):integer;  stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_SetAcessAntenna(hr:longword;AntennaID:byte):integer;  stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_OpenRFTransmitter(hr:longword):integer;  stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_CloseRFTransmitter(hr:longword):integer;  stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_SetCommuImmeTimeout(hr:longword):integer;  stdcall ; external 'rfidlib_reader.dll' ;
function   RDR_ResetCommuImmeTimeout(hr:longword):integer;stdcall;external 'rfidlib_reader.dll' ;
function   RDR_GetAntennaInterfaceCount(hr:longword):longword;stdcall ;external 'rfidlib_reader.dll'
function   RDR_GetReaderInfor(hr:longword;infType:Byte;buf:pchar;var nSize :longword) :integer;  stdcall ; external 'rfidlib_reader.dll' ;


function  RDR_CreateSetOutputOperations():longword ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_AddOneOutputOperation( hOperas:longword; outNo:byte ; outMode:byte; outFrequency:longword; outActiveDuration:longword;  outInactiveDuration:longword) :integer;  stdcall ; external 'rfidlib_reader.dll' ;
function  RDR_SetOutput( hr:longword ;outputOpers:longword):integer ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_LoadFactoryDefault(hr:longword ):integer ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_SystemReset(hr:longword ):integer ; stdcall ;external 'rfidlib_reader.dll'
function  RDR_IsSupportInventoryAsyncOutput(hr:longword;var suppported:byte):integer ; stdcall ;external 'rfidlib_reader.dll'

function RDR_ConfigBlockWrite(hr:longword; cfgno :longword ;cfgdata:PBYTE ; nSize:longword ; mask:longword) :integer ; stdcall ;external 'rfidlib_reader.dll'
function RDR_ConfigBlockRead(hr:longword; cfgno:longword ;cfgbuff:PBYTE ; nSize:longword):integer ; stdcall ;external 'rfidlib_reader.dll'
function RDR_ConfigBlockSave(hr:longword; cfgno:longword):integer ; stdcall ;external 'rfidlib_reader.dll'

function RDR_CreateRS485Node( hr:longword;busAddr:longword ;var ohrRS485Node:longword)  :integer ; stdcall ;external 'rfidlib_reader.dll'

function RDR_GetSupportedAirInterfaceProtocol(hr:longword; index:longword ;var AIPType:longword):integer ; stdcall ;external 'rfidlib_reader.dll'

function RDR_DetectNoise(hr:longword; noiseData:PBYTE ;var nSize:longword) :integer ; stdcall ;external 'rfidlib_reader.dll'
function RDR_GetGPICount(hr:longword;var oCount:longword) :integer ; stdcall ;external 'rfidlib_reader.dll'


function  RDR_ExeSpecialControlCmd(hr:longword; cmd:string  ; params:string; result:pchar ; var nSize:longword) :integer ; stdcall ;external 'rfidlib_reader.dll'

function  RDR_GetGPIsStatus(hr:longword ;
									 GPIs:PBYTE  ;
									numToGet:word  ;
									status:PBYTE) :integer ; stdcall ;external 'rfidlib_reader.dll'

function DNODE_Destroy( dn:longword):integer;stdcall;external 'rfidlib_reader.dll'

function Bluetooth_Enum():integer;stdcall;external 'rfidlib_reader.dll'
function Bluetooth_GetEnumItem(idx:integer;infType:integer; nameBuf:pchar; var nSize:longword):integer;stdcall;external 'rfidlib_reader.dll'
function RDR_SetSystemTime(hr:longword;year:longword;month:byte;day:byte;hour:byte;minute:byte;second:byte):integer;stdcall;external 'rfidlib_reader.dll'

function RDR_BuffMode_FetchRecords(hr:longword;flags:longword) :integer;stdcall;external 'rfidlib_reader.dll'

function RDR_ParseTagDataReportRaw(hTagReport:longword;rawBuffer:pbyte; var nSize:longword):integer;stdcall;external 'rfidlib_reader.dll'

function RDR_BuffMode_ClearRecords(hr:longword):integer;stdcall;external 'rfidlib_reader.dll';

implementation

end.

File: reference/rfid.sdk.win.V02.09/rfid_meeting_gate/samples/java/src/RFID/rfidlib_reader.java
package RFID;

public class rfidlib_reader {
	/*********************************functions opened*****************************************************/
	static String osName = "";
	public static void LoadLib(String sLibPath, int osType, int arType)
	{
		String libPath = "";
		
		String architecture = "";
		if (osType == rfid_def.VER_LINUX)
		{
			osName = "Linux";
		}
		else if (osType == rfid_def.VER_WINDOWS)
		{
			osName = "Windows";
		}

		if (arType == rfid_def.AR_X86)
		{
			architecture = "x86";
		}
		else if (arType == rfid_def.AR_X64)
		{
			architecture = "x64";
		}

		if (osName.equals("Windows"))
		{
			libPath = String.format("%s/libs/%s/%s/rfidlib_reader.dll",
					sLibPath, osName, architecture);
			System.load(libPath);
			libPath = String.format("%s/libs/%s/%s/jni_rfidlib_reader.dll",
					sLibPath, osName, architecture);
			System.load(libPath);
			
			libPath = String.format("%s/libs/%s/%s/Drivers",
					sLibPath, osName, architecture);
			RDR_LoadReaderDrivers(libPath);
		}
		else if (osName.equals("Linux"))
		{
		}
	}
	public native static int RDR_GetLibVersion(char[] buffer ,int nSize);
	public native static int RDR_LoadReaderDrivers(String path) ;
	public native static int RDR_GetLoadedReaderDriverCount() ;
	public native static int RDR_GetLoadedReaderDriverOpt(int idx ,String option ,char[] valueBuffer,Integer nSize/**out**/);
	public native static int RDR_GetLoadedReaderDriverOptByName(String ID ,String option ,char[] valueBuffer,Integer nSize/***out**/) ;
	public native static int HID_Enum(String DevId)  ;
	public native static int HID_GetEnumItem(int indx ,byte infType,char[] infBuffer,Integer nSize/**out**/);
	public native static int COMPort_Enum() ;
	public native static int COMPort_GetEnumItem(int idx,char[] nameBuf,Integer nSize/*out*/) ;
	public native static int RDR_Open(String connStr ,Long hrOut/*out*/);
	public native static int RDR_Close(long hr);
	public native static int RDR_CreateInvenParamSpecList()  ;  
	public native static int RDR_TagInventory(long hr,byte AIType,byte AntennaCount,byte[]AntennaIDs,int InvenParamSpecList);
	public native static int RDR_GetTagDataReportCount(long hr);
	public native static int RDR_GetTagDataReport(long hr , byte seek);
	public native static int RDR_TagConnect(long hr,int ConnParms,Integer hTag/*out*/);
	public native static int RDR_TagDisconnect(long hr,long hTag)  ;
	public native static int RDR_DisconnectAllTags(long hr)  ;
	public native static int RDR_GetReaderLastReturnError(long hr)  ;
	public native static int RDR_SetAcessAntenna(long hr ,byte AntennaID) ;
    public native static int RDR_OpenRFTransmitter(long hr) ;
	public native static int RDR_CloseRFTransmitter(long hr) ;
	public native static int RDR_SetCommuImmeTimeout(long hr) ;
	public native static int RDR_ResetCommuImmeTimeout(long hr);
	public native static int RDR_GetAntennaInterfaceCount(long hr)  ;
	public native static int RDR_GetOutputCount(long hr,Byte nCount);
	public native static int RDR_GetOutputName(long hr,byte idxOut,char[] bufName ,Integer nSize);
	public native static long RDR_CreateSetOutputOperations() ;
	public native static int RDR_AddOneOutputOperation(long hOperas,byte outNo,byte outMode,int outFrequency,int outActiveDuration ,int outInactiveDuration) ;
	public native static int RDR_SetOutput(long hr ,long outputOpers) ;
	public native static int RDR_LoadFactoryDefault(long hr) ;
	public native static long RDR_CreateSetGetConfigItemList()  ;
	public native static int RDR_SystemReset(long hr) ;
	public native static int RDR_IsSupportInventoryAsyncOutput(long hr,Boolean suppported);
	public native static int RDR_GetReaderInfor(long hr,byte Type ,char[] buffer,Integer nSize);
	public native static int RDR_ConfigBlockWrite(long hr,int cfgno ,byte[] cfgdata,int nSize,int mask) ;
	public native static int RDR_ConfigBlockRead(long hr,int cfgno,byte[]cfgbuff,int nSize);
	public native static int RDR_ConfigBlockSave(long hr,int cfgno) ;
	public native static int RDR_CreateRS485Node(long hr ,int busAddr,Integer ohrRS485Node);
	public native static int RDR_GetSupportedAirInterfaceProtocol(long hr ,int index,Integer AIPType);
	public native static int RDR_GetGPICount(long hr,Integer oCount);
	public native static int RDR_LoadAuthKey(long hr ,byte storeType,byte keyType ,byte[] key,int keylen);
	public native static int RDR_SelectAuthKey(long hr ,byte keyType) ;
	public native static int RDR_SafeAuthencate(long hr,int hTag,int params);
	public native static int RDR_ResetRF(long hr);
	
	public native static int DNODE_Destroy(long hr);
	
	//设置回调函数
	//public native static int RDR_SetEventHandler(long hr,byte eventType,int i,int msg ,int hwnd,RFID_EVENT_CALLBACK_NEW cb,long param);
	//重置回调函数
	//public native static int RDR_ResetEventHandler(RFID_EVENT_CALLBACK_NEW cb,byte evenType) ;
	
	public native static int RDR_BuffMode_FetchRecords(long hr ,int flags)  ;
	public native static int RDR_BuffMode_ClearRecords(long hr);
	public native static int RDR_ParseTagDataReportRaw(long hTagReport,byte rawBuffer[],Integer nSize) ;
}
